babi/prompt.py

```diff
@@ -95,6 +105,17 @@ def _delete(self) -> None:
     def _cut_to_end(self) -> None:
         self._s = self._s[:self._x]
 
+    def _file_complete(self) -> None:
+        partial = self._s[:self._x]
```

Comment:
this doesn't seem quite right -- it's going to take imagine foo.txt is on disk and you've written `fo][od` and you tab you're going to get `foo.txtod` which is nonsensical.  I think it should really only listen to tab if you're at the end of the string

Reply:
Sometimes, this is what I want, and I use this with tools that allow it.
Example: `so][/foo.txt` -> `someLongFolder/foo.txt`

I understand this is a question of preference, I'd personally prefer this behaviour to stay as is.

[I usually add a space to the right of the cursor before hitting tab for visual separation and then delete it after completion has done its job, but that uses the same mechanic. I doubt we'd want to check for whitespace after the cursor and make it conditional]

Reply:
in that case it could listen to tab if it is at the end of the string or before a path separator I guess?